include/ffi.h.in

```diff
@@ -478,9 +477,18 @@ ffi_status ffi_get_struct_offsets (ffi_abi abi, ffi_type *struct_type,
 #define FFI_TYPE_STRUCT     13
 #define FFI_TYPE_POINTER    14
 #define FFI_TYPE_COMPLEX    15
+#if @HAVE_EXT_VECTOR@
+#define FFI_TYPE_EXT_VECTOR 16
+#else
+#define FFI_TYPE_EXT_VECTOR FFI_TYPE_STRUCT
```

Comment:
I guess I'm not clear on why you are defining FFI_TYPE_EXT_VECTOR when ! HAVE_EXT_VECTOR.

Reply:
FFI_TYPE_EXT_VECTOR is necessary for distinguishing between arrays/struct and simd types which use NEON registers in arm64. For other architectures FFI_TYPE_STRUCT handles simd types.

Reply:
If a new type is needed, then I think there should be a patch to the documentation as well.